security/s2a/internal/crypter/common.go

```diff
@@ -23,3 +28,40 @@ func sliceForAppend(in []byte, n int) (head, tail []byte) {
 	tail = head[len(in):]
 	return head, tail
 }
+
+// encrypt is the encryption function for an AEAD crypter. aead determines
+// the type of AEAD crypter. dst can contain bytes at the beginning of
+// the ciphertext that will not be encrypted but will be authenticated. If dst
+// has enough capacity to hold these bytes, the ciphertext and the tag, no
+// allocation and copy operations will be performed. dst and plaintext may
+// fully overlap or not at all.
+func encrypt(aead cipher.AEAD, dst, plaintext, nonce, aad []byte) ([]byte, error) {
+	if len(nonce) != nonceSize {
+		return nil, fmt.Errorf("nonce size must be %d bytes. received: %d", nonceSize, len(nonce))
+	}
+	// If we need to allocate an output buffer, we want to include space for
+	// the tag to avoid forcing the caller to reallocate as well.
+	dlen := len(dst)
+	dst, out := sliceForAppend(dst, len(plaintext)+tagSize)
+	data := out[:len(plaintext)]
+	copy(data, plaintext) // data may fully overlap plaintext
+
+	// Seal appends the ciphertext and the tag to its first argument and
+	// returns the updated slice. However, sliceForAppend above ensures that
+	// dst has enough capacity to avoid a reallocation and copy due to the
+	// append.
+	dst = aead.Seal(dst[:dlen], nonce, data, aad)
+	return dst, nil
+}
+
+func decrypt(aead cipher.AEAD, dst, ciphertext, nonce, aad []byte) ([]byte, error) {
```

Comment:
I think we're missing a comment describing what `decrypt` does, similar to what exists for `encrypt`.

Reply:
Done

Reply:
Please add comments that describe the other arguments as well (dst, ciphertext, nonce, aad) similar to what exists for `Encrypt`.

Also, the [ALTS version does not have comments on `Decrypt`](https://github.com/grpc/grpc-go/blob/master/credentials/alts/internal/conn/aes128gcm.go), so once we agree on the wording here you can send a PR to @cesarghali that adds the comment there as well.